src/data/actions/assets.js

```diff
@@ -1,31 +1,47 @@
 import * as clientApi from '../api/client';
 import { assetActions } from '../constants/actionTypes';
 
+const isCorrectResponse = (request, lastRequest) => (
+  request.page === lastRequest.page &&
+  request.sort === lastRequest.sort &&
+  request.direction === lastRequest.direction &&
+  request.assetTypes.Audio === lastRequest.assetTypes.Audio &&
+  request.assetTypes.Code === lastRequest.assetTypes.Code &&
+  request.assetTypes.Documents === lastRequest.assetTypes.Documents &&
+  request.assetTypes.Images === lastRequest.assetTypes.Images &&
+  request.assetTypes.OTHER === lastRequest.assetTypes.OTHER
+);
+
 export const requestAssetsSuccess = response => ({
   type: assetActions.REQUEST_ASSETS_SUCCESS,
-  data: response.assets,
+  data: response,
 });
 
 export const assetDeleteFailure = response => ({
   type: assetActions.DELETE_ASSET_FAILURE,
   response,
 });
 
-export const getAssets = (assetsParameters, courseDetails) =>
-  dispatch =>
+export const getAssets = (request, courseDetails) =>
+  (dispatch, getState) =>
     clientApi.requestAssets(courseDetails.id, {
-      page: assetsParameters.page,
-      assetTypes: assetsParameters.assetTypes,
-      sort: assetsParameters.sort,
-      direction: assetsParameters.direction,
+      page: request.page,
+      assetTypes: request.assetTypes,
+      sort: request.sort,
+      direction: request.direction,
     })
       .then((response) => {
         if (response.ok) {
           return response.json();
         }
         throw new Error(response);
       })
-      .then(json => dispatch(requestAssetsSuccess(json)))
+      .then((json) => {
+        const lastRequest = getState().request;
```

Comment:
`lastRequest` might not be clear. It's a little difficult to describe what situation this is solving, since both requests are actual requests. lastRequest is the last request that was recorded as necessary to make based on changes to the UI state, whereas request refers to the request in this current getAssets call. Does that make sense?

Reply:
I think this make sense. I was trying to think about other words too, such as "most recent" or "current", but "last" seems just fine!

Reply:
Thanks!